libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Generators/TemplateEngineLanguageGenerator.cs

```diff
@@ -67,6 +70,7 @@ public TemplateEngineLanguageGenerator(string filePath, Dictionary<string, IList
             var (_, locale) = LGResourceLoader.ParseLGFileName(Id);
             var importResolver = LanguageGeneratorManager.ResourceExplorerResolver(locale, resourceMapping);
             this.lg = LanguageGeneration.Templates.ParseFile(filePath, importResolver);
+            this.lg.AddDebuggingEventRegister(SourceMapRegister);
```

Comment:
i don't think we should require a callback just for registering sourcemap, a callback function should only be required when do evalutation. 

If certain sub-structure is not not available after parseFile, we should extend that, to make it available. 

The principle is after parsing, i should have enough information to register source map.